src/api/hooks/channel.ts

```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@apollo/client'
+import { GET_CHANNEL } from '@/api/queries/channels'
+import { GetChannel, GetChannelVariables } from '@/api/queries/__generated__/GetChannel'
+
+export const useChannel = (id: string) => {
+  const { data, loading, error } = useQuery<GetChannel, GetChannelVariables>(GET_CHANNEL, {
+    variables: { id },
+  })
+  return {
+    loading,
+    data: data?.channel,
+    error,
+  }
+}
+
+export const useChannelLink = (id: string | undefined) => {
```

Comment:
`ChannelLink` isn't a resource, so we shouldn't have a separate query for that. The purpose here is to `useChannel` but with some additional options. What you can do is add something like `opts` arg to `useChannel` that will allow specifying things like `fetchPolicy` and `skip`